lib/jormungandr/test/integration/Cardano/WalletSpec.hs

```diff
@@ -0,0 +1,139 @@
+    ( BaseUrl (..), Jormungandr, Scheme (..) )
+import Cardano.Wallet.Jormungandr.Environment
+    ( Network (..) )
+import Cardano.Wallet.Jormungandr.Network
+    ( JormungandrLayer (..)
+    , defaultManagerSettings
+    , getInitialBlockchainParameters
+    , newManager
+    )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Passphrase (..), digest, generateKeyFromSeed, publicKey )
+import Cardano.Wallet.Primitive.AddressDiscovery
+    ( mkSeqState )
+import Cardano.Wallet.Primitive.Mnemonic
+    ( EntropySize, entropyToBytes, genEntropy )
+import Cardano.Wallet.Primitive.Model
+    ( currentTip )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..)
+    , Hash (..)
+    , SlotId (..)
+    , WalletId (..)
+    , WalletName (..)
+    )
+import Cardano.Wallet.Unsafe
+    ( unsafeRunExceptT )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( Async, async, cancel )
+import Control.Monad
+    ( unless, void )
+import Data.Coerce
+    ( coerce )
+import Data.Time.Clock
+    ( addUTCTime, getCurrentTime )
+import System.Directory
+    ( removePathForcibly )
+import Test.Hspec
+    ( Spec, after, before, expectationFailure, it )
+
+import qualified Cardano.Wallet.DB.MVar as MVar
+import qualified Cardano.Wallet.Jormungandr.Network as Jormungandr
+import qualified Data.ByteString as BS
+
+spec :: Spec
+spec = do
+    before startNode $ after killNode $ do
+        it "A newly created wallet can sync with the chain" $ \(_, wallet) -> do
+            bytes <- entropyToBytes <$> genEntropy @(EntropySize 15)
+            let xprv = generateKeyFromSeed (Passphrase bytes, mempty) mempty
+            wid <- unsafeRunExceptT $ createWallet wallet
+                (WalletId $ digest $ publicKey xprv)
+                (WalletName "My Wallet")
+                (mkSeqState @(Jormungandr 'Testnet) (xprv, mempty) minBound)
+            unsafeRunExceptT $ restoreWallet wallet wid
+            threadDelay 2000000
+            tip <- slotId . currentTip . fst <$>
+                unsafeRunExceptT (readWallet wallet wid)
+            unless (tip > (SlotId 0 0)) $
+                expectationFailure ("The wallet tip is still " ++ show tip)
+
+        it "The time of the tip is recent" $ \(_, wallet) -> do
+            bytes <- entropyToBytes <$> genEntropy @(EntropySize 15)
+            let xprv = generateKeyFromSeed (Passphrase bytes, mempty) mempty
+            wid <- unsafeRunExceptT $ createWallet wallet
+                (WalletId $ digest $ publicKey xprv)
+                (WalletName "My Wallet")
+                (mkSeqState @(Jormungandr 'Testnet) (xprv, mempty) minBound)
+            unsafeRunExceptT $ restoreWallet wallet wid
+            threadDelay 8000000
+            tip <- slotId . currentTip . fst <$>
+                unsafeRunExceptT (readWallet wallet wid)
+            let mintedTime = slotIdTime wallet tip
+            t <- getCurrentTime
+            if t > (addUTCTime (-20) mintedTime)
```

Comment:
This number looks quite arbitrary, what's the rationale here?

Reply:
The maximum delay should be `SlotDuration = 10 s`, so instead making sure the tip was minted within 20 seconds ago is just for some additional marginal.

Reply:
Would be good to make this explicit. Especially because the slot duration isn't a "global constant" that necessarily has the same value. We know it's `10` because we set it to `10` in the genesis block. The best would be to expose a `getBlockchainParameters` from the wallet layer and use the slot duration provided there.

Also, if the maximum delay is the slot duration, we should check for the slot duration, and not twice of it.